public/docs/dart/latest/guide/dependency-injection.jade

```diff
@@ -1,6 +1,782 @@
+  system which means nested injectors can create their own service instances.
+  Learn more about that in the [Hierarchical Injection](./hierarchical-dependency-injection.html) chapter.
+
+  ### Testing the component
+  We emphasized earlier that designing a class for dependency injection makes the class easier to test.
+  Listing dependencies as constructor parameters may be all we need to test application parts effectively.
+
+  For example, we can create a new `HeroListComponent` with a mock service that we can manipulate
+  under test:
+
++makeExample('dependency-injection/dart/web/test.component.dart', 'spec')(format='.')
+.l-sub-section
+  :marked
+    Learn more in the [Testing](../testing/index.html) chapter.
+
+:marked
+  ### When the service needs a service
+  Our `HeroService` is very simple. It doesn't have any dependencies of its own.
+
+
+  What if it had a dependency? What if it reported its activities through a logging service?
+  We'd apply the same *constructor injection* pattern,
+  adding a constructor that takes a `logger` parameter.
+
+  Here is the revision compared to the original.
+
++makeTabs(
+`dependency-injection/dart/web/heroes/hero.service.2.dart,
+dependency-injection/dart/web/heroes/hero.service.1.dart`,
+null,
+`app/heroes/hero.service (v.2),
+app/heroes/hero.service (v.1)`)
+
+:marked
+  The constructor now asks for an injected instance of a `Logger` and stores it in a private property called `_logger`.
+  We call that property within our `getHeroes()` method when anyone asks for heroes.
+
+  <a id="injectable"></a>
+  ### Why *@Injectable*?
+  Notice the `@Injectable()` decoration above the service class.
+  We haven't seen `@Injectable()` before.
+  As it happens, we could have added it to our first version`HeroService`.
+  We didn't bother because we didn't need it then.
+
+  We need it now... now that our service has an injected dependency.
+  We need it because Angular requires constructor parameter metadata in order to inject a `Logger`.
+  As [we mentioned earlier](#di-metadata), Dart *only generates metadata for classes that have a decorator*. .
+
+.callout.is-helpful
+  header Always add @Injectable()
+  :marked
+    We recommend adding `@Injectable()` to every service class, even those that don't have dependencies
+    and, therefore, do not technically require it. Two reasons:
+  ol(style="font-size:inherit")
+    li <i>Future proofing</i>: no need to remember <code>@Injectable</code> when we add a dependency later.
+    li <i>Consistency</i>: all services follow the same rules and we don't have to wonder why a decorator is missing.
+.l-sub-section
+  :marked
+    The `HeroesComponent` has an injected dependency too. Why don't we add `@Injectable()` to the `HeroesComponent`?
+
+    We *can* add it if we really want to. It isn't necessary because the `HeroesComponent` is already decorated with `@Component`.
+    Dart generates metadata for *any* class with a decorator and *any* decorator will do.
```

Comment:
That doesn't seem right. Only a specified list of annotations are considered for Angular DI AFAIK.